server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -98,23 +99,38 @@ public ConnectionProfile getConnectionProfile() {
     }
 
     public Transport.Connection getAnyRemoteConnection() {
-        List<Transport.Connection> localConnections = this.connections;
-        if (localConnections.isEmpty()) {
-            throw new NoSuchRemoteClusterException(clusterAlias);
-        } else {
-            long curr;
-            while ((curr = counter.incrementAndGet()) == Long.MIN_VALUE);
-            return localConnections.get(Math.floorMod(curr, localConnections.size()));
+        List<DiscoveryNode> localConnectedNodes = this.connectedNodes;
+        long curr;
+        while ((curr = counter.incrementAndGet()) == Long.MIN_VALUE);
+        if (localConnectedNodes.isEmpty() == false) {
+            DiscoveryNode nextNode = localConnectedNodes.get(Math.floorMod(curr, localConnectedNodes.size()));
+            try {
+                return delegate.getConnection(nextNode);
+            } catch (NodeNotConnectedException e) {
+                // Ignore. We will manually create an iterator of open nodes
+            }
+        }
+        if (delegate.size() != 0) {
```

Comment:
NIT: Why have this check? Seems kind of racy and not much of an optimization if any to begin with?